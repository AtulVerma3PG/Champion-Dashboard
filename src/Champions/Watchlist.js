import React from "react";
import { Navbar } from "react-bootstrap";
import { useHistory, Link } from "react-router-dom";
import * as ReactBootStrap from "react-bootstrap";
import "./Css/Champion.css";

/**
 * This Function is responsible to show the Watchlist
 *
 * @param {object} props Watchlist details
 */
const Watchlist = () => {
  const state = JSON.parse(localStorage.getItem("state"));
  const { watchlist } = state;
  const history = useHistory();
  const openHome = () => {
    history.push("/");
  };

  /**
   * Update Local storage on state change
   */
  const removeChampion = (event) => {
    state.watchlist = watchlist.filter((e) => e.id !== event);
    localStorage.setItem("state", JSON.stringify(state));
    window.location.reload(false);
  };

  return (
    <div>
      <Navbar
        bg="dark"
        variant="dark"
      >
        <Navbar.Brand>My Watchlist</Navbar.Brand>
        <button
          type="button"
          className="btn btn-primary btn-lg"
          onClick={openHome}
        >
          Home
        </button>
      </Navbar>
      <ReactBootStrap.Table responsive="md" bordered hover>
        <thead>
          <tr className="center">
            <th>Champion Profile</th>
            <th>Champion ID</th>
            <th>Name</th>
            <th>Armor</th>
            <th>armorperlevel</th>
            <th>hp</th>
            <th>Watchlist Action</th>
          </tr>
        </thead>
        <tbody>
          {watchlist.map((champion) => (
            <tr key={champion.id} className="center">
              <td>
                <img src={champion.image_url} alt={champion.image_url} />
              </td>
              <td>{champion.id}</td>
              <td>
                <Link
                  to={{
                    pathname: "/ChampionDetails",
                    state: { champion },
                  }}
                >
                  {champion.name}
                </Link>
              </td>
              <td>{champion.armor}</td>
              <td>{champion.armorperlevel}</td>
              <td>{champion.hp}</td>
              <td>
                <div>
                  <button
                    type="button"
                    className="btn btn-primary btn-lg"
                    onClick={() => removeChampion(champion.id)}
                  >
                    Remove
                  </button>
                </div>
              </td>
            </tr>
          ))}
        </tbody>
      </ReactBootStrap.Table>
    </div>
  );
};

export default Watchlist;
