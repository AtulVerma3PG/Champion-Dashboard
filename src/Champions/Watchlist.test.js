/* eslint-disable */
import React from "react";
import renderer from "react-test-renderer";
import { createMemoryHistory } from "history";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure, mount } from "enzyme";
import Watchlist from "./Watchlist";

const watchlist = [];

test("First Snapshot test", () => {
  localStorage.setItem("watchlist", JSON.stringify(watchlist));
  const component = renderer.create(
    <Watchlist/>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  localStorage.removeItem("watchlist");
});

configure({ adapter: new Adapter() });
