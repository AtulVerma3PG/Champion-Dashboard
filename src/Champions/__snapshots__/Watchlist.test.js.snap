// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First Snapshot test 1`] = `
<div>
  <nav
    className="navbar navbar-expand navbar-dark bg-dark"
  >
    <span
      className="navbar-brand"
    >
      My Watchlist
    </span>
    <button
      className="btn btn-primary btn-lg"
      onClick={[Function]}
      type="button"
    >
      Home
    </button>
  </nav>
  <div
    className="table-responsive-md"
  >
    <table
      className="table table-bordered table-hover"
    >
      <thead>
        <tr>
          <th>
            Champion Profile
          </th>
          <th>
            Champion ID
          </th>
          <th>
            Name
          </th>
          <th>
            Armor
          </th>
          <th>
            armorperlevel
          </th>
          <th>
            hp
          </th>
          <th>
            Watchlist Action
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
